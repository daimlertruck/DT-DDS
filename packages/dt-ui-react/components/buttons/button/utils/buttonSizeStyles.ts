import { CustomTheme as Theme } from '../../../../types';
import { ButtonSizes } from '../constants/types';

export const buttonSizeStyles = (size: ButtonSizes, theme: Theme) =>
  ({
    large: `
      ${theme.fontStyles.button1}
      padding: ${theme.spacing['2xs']} ${theme.spacing.s};
    `,
    medium: `
      ${theme.fontStyles.button2}
      padding: ${theme.spacing['3xs']} ${theme.spacing.xs};
    `,
    small: `
      ${theme.fontStyles.button2}
      padding: ${theme.spacing['4xs']} ${theme.spacing['2xs']};
    `,
  }[size]);
