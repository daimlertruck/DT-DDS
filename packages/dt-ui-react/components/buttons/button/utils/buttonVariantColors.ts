import { CustomTheme as Theme } from '../../../../types';
import {
  ButtonVariant,
  ContainedColors,
  OutlinedColors,
  TextColors,
} from '../constants/types';

interface ColorKeys {
  color: string;
  bgColor: string;
  hoverBgColor?: string;
  hoverColor?: string;
}

export const buttonVariantColors = (
  color: ContainedColors | OutlinedColors,
  theme: Theme
): Record<ButtonVariant, ColorKeys> => ({
  contained: {
    color: theme.colors.alpha.white,
    bgColor: theme.palette.alpha[color],
    hoverBgColor: theme.palette.alpha[`${color}Dark_300` as ContainedColors],
  },
  outlined: {
    color:
      color === 'neutral'
        ? theme.colors.alpha.gray_200
        : theme.palette.alpha[color],
    bgColor: theme.colors.alpha.white,
    hoverBgColor: theme.palette.alpha[`${color}Light_300` as OutlinedColors],
  },
  text: {
    color: theme.palette.alpha[color],
    bgColor: 'transparent',
    hoverColor: theme.palette.alpha[`${color as TextColors}Dark_300`],
  },
});
