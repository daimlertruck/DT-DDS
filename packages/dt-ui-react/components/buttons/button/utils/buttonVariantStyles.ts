import { CustomTheme as Theme } from '../../../../types';
import {
  ButtonVariant,
  ContainedColors,
  OutlinedColors,
} from '../constants/types';

export const buttonVariantStyles = (
  variant: ButtonVariant,
  theme: Theme,
  color: ContainedColors | OutlinedColors
) =>
  ({
    outlined: `
      border: 1px solid;
      text-transform: uppercase;
        ${
          color === 'neutral'
            ? theme.palette.neutralDark_100
            : theme.palette[color]
        };

      &:disabled {
        color: ${
          color === 'neutral'
            ? theme.palette.neutralLight_100
            : theme.palette[color]
        };
        background-color: ${theme.colors.white};
        border-color: ${
          color === 'neutral'
            ? theme.palette.neutralLight_100
            : theme.palette[color]
        };
        cursor: default;
      }
    `,
    contained: `
      border: ${theme.spacing.none};
      text-transform: uppercase;
      &:disabled {
        opacity: 0.4;
        cursor: default;
      }
    `,
    text: `
      border: none;

      &:disabled {
        color: ${theme.palette.neutralLight_100};
      }
    `,
  }[variant]);
