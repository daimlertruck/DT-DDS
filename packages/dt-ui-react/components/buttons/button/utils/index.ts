export * from './buttonVariantColors';
export * from './buttonVariantStyles';
