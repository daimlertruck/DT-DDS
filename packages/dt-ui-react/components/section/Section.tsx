import React, { ReactNode } from 'react';

import { BaseProps } from '../../types';
import { Box } from '../box';
import { Spinner } from '../spinner';
import { Typography } from '../typography';

import { SectionVariant } from './constants';
import {
  DetailStyled,
  SectionStyled,
  SubSectionStyled,
  HeaderStyled,
  ContainerStyled,
  FooterStyled,
} from './Section.styled';

export interface SectionProps extends BaseProps {
  variant?: SectionVariant;
  withPageHeader?: boolean;
}

export interface SectionContentProps extends BaseProps {
  isLoading?: boolean;
}

export interface DetailProps extends BaseProps {
  children: ReactNode;
  title: string;
  isInline?: boolean;
}

export const Section = ({
  children,
  dataTestId,
  style,
  variant = SectionVariant.Box,
  withPageHeader = false,
}: SectionProps) => {
  return (
    <SectionStyled
      variant={variant}
      withPageHeader={withPageHeader}
      style={style}
      data-testid={`${dataTestId ? `${dataTestId}-` : ''}section`}
    >
      {children}
    </SectionStyled>
  );
};

Section.Header = ({ children }: BaseProps) => {
  return <HeaderStyled>{children}</HeaderStyled>;
};

Section.Title = ({ children, dataTestId }: BaseProps) => {
  return (
    <Typography element='h2' fontStyles='h5' dataTestId={dataTestId}>
      {children}
    </Typography>
  );
};

Section.Content = ({ isLoading, children }: SectionContentProps) => {
  if (isLoading) {
    return (
      <Box style={{ margin: '30px 0px' }}>
        <Spinner />
      </Box>
    );
  }

  return <>{children}</>;
};

Section.Detail = ({
  title,
  children,
  isInline = false,
  dataTestId,
  style,
}: DetailProps) => {
  return (
    <DetailStyled
      data-testid={dataTestId ?? 'section-detail'}
      isInline={isInline}
      style={style}
    >
      <Typography fontStyles='pXXSmall' color='gray_100'>
        {title}
      </Typography>
      {children}
    </DetailStyled>
  );
};

Section.SubSection = ({ children, dataTestId }: BaseProps) => {
  return (
    <SubSectionStyled data-testid={dataTestId ?? 'subsection-container'}>
      {children}
    </SubSectionStyled>
  );
};

Section.Container = ({
  children,
  style,
  dataTestId = 'default',
}: SectionContentProps) => {
  return (
    <ContainerStyled
      data-testid={`${dataTestId}-section-container`}
      style={style}
    >
      {children}
    </ContainerStyled>
  );
};

Section.Footer = ({ children, dataTestId = 'default' }: BaseProps) => {
  return (
    <FooterStyled data-testid={`${dataTestId}-section-footer`}>
      {children}
    </FooterStyled>
  );
};
