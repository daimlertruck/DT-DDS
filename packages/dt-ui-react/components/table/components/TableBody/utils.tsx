import { Children, ReactNode, cloneElement, isValidElement } from 'react';

import { TableDataCellHeader } from '../TableDataCell/TableDataCell';

export const addHeadersToDataCells = (
  dataCells: ReactNode,
  headers: ReactNode[]
) => {
  return Children.map(dataCells, (dataCell, index) => {
    if (isValidElement(dataCell)) {
      const header = headers[index];
      return cloneElement(
        dataCell,
        {},
        <>
          <TableDataCellHeader>{header}</TableDataCellHeader>
          {dataCell.props.children}
        </>
      );
    }
    return dataCell;
  });
};

export const addHeadersToDataCellsInRows = (
  rows: ReactNode,
  headers: ReactNode[]
) => {
  return Children.map(rows, (row) => {
    if (isValidElement(row)) {
      return cloneElement(
        row,
        {},
        addHeadersToDataCells(row.props.children, headers)
      );
    }
    return row;
  });
};
