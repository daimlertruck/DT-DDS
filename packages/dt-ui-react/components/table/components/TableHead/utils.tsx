import { Children, ReactNode, isValidElement } from 'react';

const extractColumnHeaders = (columnHeaders: ReactNode): ReactNode[] => {
  return Children.toArray(columnHeaders).reduce<ReactNode[]>(
    (headers, columnHeader) => {
      if (isValidElement(columnHeader)) {
        return [...headers, columnHeader.props.children];
      }

      return headers;
    },
    []
  );
};

export const extractTableHeaders = (theadRow: ReactNode) => {
  if (isValidElement(theadRow)) {
    return extractColumnHeaders(theadRow.props.children);
  }

  return [];
};
