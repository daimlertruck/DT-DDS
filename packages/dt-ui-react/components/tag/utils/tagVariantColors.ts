import { CustomTheme as Theme } from '../../../types';
import { TagColor, TagVariant } from '../constants';

interface TagVariantColors {
  border: string | number;
  color: string;
  borderColor?: string;
  backgroundColor: string;
  '&:hover'?: {
    borderColor?: string;
    backgroundColor: string;
    color: string;
  };
}

interface TagVariantColorsProps {
  theme: Theme;
  color: TagColor;
  variant: TagVariant;
  hasHover: boolean;
  isDisabled: boolean;
}

const getColorWithShade = ({
  theme,
  color,
  shade,
}: Pick<TagVariantColorsProps, 'color'> & {
  shade: '90' | '60' | '30' | '10';
  theme: Theme;
}) => {
  const { colors, palette } = theme;

  if (color === 'primary') {
    return {
      '90': palette.primaryDark_500,
      '60': palette.primaryDark_200,
      '30': palette.primaryLight_100,
      '10': palette.primaryLight_400,
    }[shade];
  }

  const colorWithShade: keyof Theme['colors'] = `${color}_${shade}`;

  return colors[colorWithShade];
};

export const tagVariantColors = ({
  theme,
  color,
  variant,
  hasHover,
  isDisabled,
}: TagVariantColorsProps): TagVariantColors => {
  const colorShade90 = getColorWithShade({
    theme,
    color,
    shade: '90',
  });
  const colorShade60 = getColorWithShade({
    theme,
    color,
    shade: '60',
  });
  const colorShade30 = getColorWithShade({
    theme,
    color,
    shade: '30',
  });
  const colorShade10 = getColorWithShade({
    theme,
    color,
    shade: '10',
  });

  const colorWithDisabled = isDisabled ? colorShade30 : colorShade60;

  return {
    solid: {
      border: theme.spacing.none,
      color: theme.colors.white,
      backgroundColor: colorWithDisabled,
      ...(hasHover && {
        '&:hover': {
          backgroundColor: colorShade90,
          color: theme.colors.white,
        },
      }),
    },
    outlined: {
      border: `1px solid ${colorShade60}`,
      color: colorWithDisabled,
      backgroundColor: colorShade10,
      borderColor: colorWithDisabled,
      ...(hasHover && {
        '&:hover': {
          borderColor: colorShade90,
          backgroundColor: colorShade10,
          color: colorShade90,
        },
      }),
    },
  }[variant];
};
