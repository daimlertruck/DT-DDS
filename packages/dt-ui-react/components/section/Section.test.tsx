import { render } from '@testing-library/react';
import React from 'react';

import { withProviders } from '../../utils';
import { Typography } from '../typography';

import { Section } from './Section';

describe('<Section /> component', () => {
  const ProvidedSection = withProviders(Section);

  it('renders section component', () => {
    const { container } = render(
      <ProvidedSection>
        <Section.Header>
          <Section.Title>My Awesome Title</Section.Title>
          <p>Some header info</p>
        </Section.Header>
        <Section.SubSection>
          <Section.Detail title='Detail Title 1'>
            <Typography>First info</Typography>
          </Section.Detail>
          <Section.Detail title='Detail Title 2'>
            <Typography>Second info</Typography>
          </Section.Detail>
          <Section.Detail title='Detail Title 3' isInline={true}>
            <Typography>Third inline info</Typography>
          </Section.Detail>
        </Section.SubSection>
        <Typography element='h2' fontStyles='h5' style={{ margin: '15px 0px' }}>
          A subtitle
        </Typography>
        <p>Some text content after a subtitle</p>
      </ProvidedSection>
    );

    expect(container).toMatchSnapshot();
  });

  it('renders section component with loading content', () => {
    const { container } = render(
      <ProvidedSection>
        <Section.Header>
          <Section.Title>My Awesome Title</Section.Title>
        </Section.Header>
        <Section.Content isLoading={true}>Some content</Section.Content>
      </ProvidedSection>
    );

    expect(container).toMatchSnapshot();
  });
});

describe('<Section.Container /> component', () => {
  const ProvidedSection = withProviders(Section.Container);

  it('renders a div html element with correct content', () => {
    const { container } = render(
      <ProvidedSection>
        <Section.Header>
          <Section.Title>My Awesome Title</Section.Title>
        </Section.Header>
        <Section.Footer>Footer content</Section.Footer>
      </ProvidedSection>
    );

    expect(container).toMatchSnapshot();
  });
});
