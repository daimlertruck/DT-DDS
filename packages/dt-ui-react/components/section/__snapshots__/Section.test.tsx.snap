// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section /> component renders section component 1`] = `
.emotion-0 {
  margin-top: 32px;
  background-color: #FFFFFF;
  box-shadow: 1px 1px 3px 1px rgba(0, 0, 0, 0.1);
  padding: 32px;
}

.emotion-0>div:first-of-type {
  padding-top: 24px;
}

.emotion-2 {
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4 {
  font-weight: 400;
  font-size: 1.5rem;
  line-height: 1.23;
  margin: 0;
  color: #151617;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-8 p:first-of-type {
  display: block;
  padding-right: 0;
}

.emotion-8 p:last-child {
  display: block;
}

.emotion-10 {
  font-weight: 400;
  font-size: 0.75rem;
  line-height: 1.34;
  margin: 0;
  color: #8B8C8C;
}

.emotion-12 {
  font-weight: 400;
  font-size: 1.125rem;
  line-height: 1.34;
  margin: 0;
  color: #151617;
}

.emotion-20 p:first-of-type {
  display: inline;
  padding-right: 10px;
}

.emotion-20 p:last-child {
  display: inline;
}

<div>
  <section
    class="emotion-0 emotion-1"
    data-testid="section"
  >
    <header
      class="emotion-2 emotion-3"
    >
      <h2
        class="emotion-4 emotion-5"
        color="gray_300"
        data-testid="typography"
      >
        My Awesome Title
      </h2>
      <p>
        Some header info
      </p>
    </header>
    <div
      class="emotion-6 emotion-7"
      data-testid="subsection-container"
    >
      <div
        class="emotion-8 emotion-9"
        data-testid="section-detail"
      >
        <p
          class="emotion-10 emotion-5"
          color="gray_100"
          data-testid="typography"
        >
          Detail Title 1
        </p>
        <p
          class="emotion-12 emotion-5"
          color="gray_300"
          data-testid="typography"
        >
          First info
        </p>
      </div>
      <div
        class="emotion-8 emotion-9"
        data-testid="section-detail"
      >
        <p
          class="emotion-10 emotion-5"
          color="gray_100"
          data-testid="typography"
        >
          Detail Title 2
        </p>
        <p
          class="emotion-12 emotion-5"
          color="gray_300"
          data-testid="typography"
        >
          Second info
        </p>
      </div>
      <div
        class="emotion-20 emotion-9"
        data-testid="section-detail"
      >
        <p
          class="emotion-10 emotion-5"
          color="gray_100"
          data-testid="typography"
        >
          Detail Title 3
        </p>
        <p
          class="emotion-12 emotion-5"
          color="gray_300"
          data-testid="typography"
        >
          Third inline info
        </p>
      </div>
    </div>
    <h2
      class="emotion-4 emotion-5"
      color="gray_300"
      data-testid="typography"
      style="margin: 15px 0px;"
    >
      A subtitle
    </h2>
    <p>
      Some text content after a subtitle
    </p>
  </section>
</div>
`;

exports[`<Section /> component renders section component with loading content 1`] = `
@keyframes animation-0 {
  from {
    -webkit-transform: rotate(0);
    -moz-transform: rotate(0);
    -ms-transform: rotate(0);
    transform: rotate(0);
  }

  to {
    -webkit-transform: rotate(360deg);
    -moz-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

.emotion-0 {
  margin-top: 32px;
  background-color: #FFFFFF;
  box-shadow: 1px 1px 3px 1px rgba(0, 0, 0, 0.1);
  padding: 32px;
}

.emotion-0>div:first-of-type {
  padding-top: 24px;
}

.emotion-2 {
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4 {
  font-weight: 400;
  font-size: 1.5rem;
  line-height: 1.23;
  margin: 0;
  color: #151617;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-10 {
  width: 74px;
  height: 74px;
  border: 2px solid #CCCCCC;
  border-radius: 50%;
  border-top: 2px solid #0078D6;
  -webkit-animation: animation-0 1s cubic-bezier(0.3, 0.1, 0.25, 0.65) 0s infinite normal none;
  animation: animation-0 1s cubic-bezier(0.3, 0.1, 0.25, 0.65) 0s infinite normal none;
}

<div>
  <section
    class="emotion-0 emotion-1"
    data-testid="section"
  >
    <header
      class="emotion-2 emotion-3"
    >
      <h2
        class="emotion-4 emotion-5"
        color="gray_300"
        data-testid="typography"
      >
        My Awesome Title
      </h2>
    </header>
    <div
      class="emotion-6 emotion-7"
      style="margin: 30px 0px;"
    >
      <div
        class="emotion-6 emotion-7"
        style="margin: auto;"
      >
        <div
          class="emotion-10 emotion-11"
        />
      </div>
    </div>
  </section>
</div>
`;

exports[`<Section.Container /> component renders a div html element with correct content 1`] = `
.emotion-0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  border: 1px solid #CCCCCC;
  border-top: none;
}

.emotion-2 {
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4 {
  font-weight: 400;
  font-size: 1.5rem;
  line-height: 1.23;
  margin: 0;
  color: #151617;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-item: center;
  -ms-flex-item-align-item: center;
  align-item: center;
  width: 100%;
  padding: 16px 32px;
  background-color: #F4F4F4;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-testid="default-section-container"
  >
    <header
      class="emotion-2 emotion-3"
    >
      <h2
        class="emotion-4 emotion-5"
        color="gray_300"
        data-testid="typography"
      >
        My Awesome Title
      </h2>
    </header>
    <div
      class="emotion-6 emotion-7"
      data-testid="default-section-footer"
    >
      Footer content
    </div>
  </div>
</div>
`;
